### PR TITLE
Solved level3

### PR DESCRIPTION
**Pull request by iuliagatea**

### CODE REVIEW

File: ruby_proj/app/cart.rb [hunk 1]
```diff
@@ -13,8 +13,24 @@ def total
...
 
+  def cart_items_with_qty
+    items.map(&:to_hash_with_qty)
+  end
+
+  def qty_count
+    items.map(&:qty).inject(0, :+)
+  end
+
+  def check_item(args = {})
```
Comment (asarbu) [inline ruby_proj/app/cart.rb]: suggestion: `find`

File: ruby_proj/app/bookstore.rb [hunk 2]
```diff
@@ -16,9 +16,15 @@ def sort!
   end
 
   def add_to_cart(title)
-    book = find_by(title)
-    item = Item.new(cart.item_count, book)
-    cart.items << item
+    check_item = cart.check_item(title)
+    if check_item.nil?
```
Comment (asarbu) [inline ruby_proj/app/bookstore.rb]: suggestion: `item.present?`

File: ruby_proj/app/bookstore.rb [hunk 3]
```diff
@@ -16,9 +16,15 @@ def sort!
...
+      book = find_by(title)
+      item = Item.new(cart.item_count, book)
+      cart.items << item
+    else
+      item = check_item
```
Comment (paucio) [inline ruby_proj/app/bookstore.rb]: This can be in one line: `check_item.qty += 1`

File: ruby_proj/app/bookstore.rb (hunk 2 above)
Comment (iuliagatea) [reply ruby_proj/app/bookstore.rb]: with item.present i have this error:
`undefined method `present?' for nil:NilClass`

File: ruby_proj/app/bookstore.rb (hunk 3 above)
Comment (iuliagatea) [reply ruby_proj/app/bookstore.rb]: done

**Pull request merged**